src/hooks/block-subscription/block-subscription.hook.ts

```diff
@@ -0,0 +1,52 @@
+import { BlockResponse, BlockFullHeader } from '@taquito/rpc';
+import { useEffect, useRef, useState } from 'react';
+
+export interface BlockInterface extends Pick<BlockResponse, 'protocol' | 'chain_id' | 'hash'> {
+  header: Pick<BlockFullHeader, 'level' | 'timestamp'>;
+}
+
+export const EMPTY_BLOCK: BlockInterface = {
+  protocol: '',
+  chain_id: '',
+  hash: '',
+  header: {
+    level: 0,
+    timestamp: ''
+  }
+};
+
+export const useBlockSubscription = (webSocketUrl: string) => {
```

Comment:
`webSocketUrl` parameter would not be reused.  
just create a local module variable 
```
const BLOCK_UPDATE_API_URL = `${TEZOS_DEXES_API_URL}/block`;
```

and reuse it